### PR TITLE
Add c formatted io table

### PR DESCRIPTION
**Pull request by mppf**
This commit adds a table "Formatted I/O for C Programmers" to the IO.chpl docs.

### CODE REVIEW

**Comment by mppf** [comment]
@bradcray could you look at this? is it what you hoped for?

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -415,6 +415,28 @@ Syntax in Detail" section below.
...
 
+Formatted I/O for C Programmers
++++++++++++++++++++++++++++++++
+
+========  ===========  ==========================================
+C         Chapel       Meaning
+========  ===========  ==========================================
+%i        %i           an integer in decimal
+%d        %i           an integer in decimal
+%x        %xu          an unsigned number in hexadecimal
```
Comment (bradcray) [inline modules/standard/IO.chpl]: I'd think that before this line you'd want a %u -> %i case?

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -415,6 +415,28 @@ Syntax in Detail" section below.
...
+%g        %r           real number in exponential or decimal if compact
+%.2g      %2.r         real number with 2 significant digits
+%f        %dr          real number always in decimal
+%.3f      %.3dr        real number in decimal with 3 digits after ``.``
+%e        %er          real number always in exponential
+%.3e      %.3er        real number in exponential with 3 digits after ``.``
```
Comment (bradcray) [inline modules/standard/IO.chpl]: The above e/f/\g forms also can take a number prior to the '.', right?  (as can %d and such, I think?)  Does it make sense to have an example of one of those as well/instead?  (e.g., "%13.5g")

**Comment by bradcray** [comment]
Yep, this exactly what I was looking for.  Putting an introductory sentence before the table like "The following table is designed to help C programmers make the conversion to Chapel format strings." might be good.  Thanks!

**Pull request merged**